packages/test/src/test-worker-versioning.ts

```diff
@@ -0,0 +1,181 @@
+    }
+    t.context = {
+      client,
+      doSkip,
+    };
+  });
+
+  test('Worker versioning workers get appropriate tasks', withSkipper, async (t) => {
+    const taskQueue = 'worker-versioning-tasks-' + uuid4();
+    const wf1Id = 'worker-versioning-1-' + uuid4();
+    const wf2Id = 'worker-versioning-2-' + uuid4();
+    const client = t.context.client;
+    await client.taskQueue.updateWorkerBuildIdCompatibility(taskQueue, {
+      operation: 'newIdInNewDefaultSet',
+      buildId: '1.0',
+    });
+
+    const worker1 = await Worker.create({
+      workflowsPath: require.resolve('./workflows'),
+      activities,
+      taskQueue,
+      showStackTraceSources: true,
+      buildId: '1.0',
+      useWorkerVersioning: true,
+    });
+    const worker1Prom = worker1.run();
+    worker1Prom.catch((err) => {
+      t.fail('Worker 1.0 run error: ' + JSON.stringify(err));
+    });
+
+    const wf1 = await client.workflow.start('unblockOrCancel', {
+      taskQueue,
+      workflowId: wf1Id,
+    });
+    await client.taskQueue.updateWorkerBuildIdCompatibility(taskQueue, {
+      operation: 'newIdInNewDefaultSet',
+      buildId: '2.0',
+    });
+    const wf2 = await client.workflow.start('unblockOrCancel', {
+      taskQueue,
+      workflowId: wf2Id,
+    });
+    await wf1.signal(unblockSignal);
+
+    const worker2 = await Worker.create({
+      workflowsPath: require.resolve('./workflows'),
+      activities,
+      taskQueue,
+      showStackTraceSources: true,
+      buildId: '2.0',
+      useWorkerVersioning: true,
+    });
+    const worker2Prom = worker2.run();
+    worker2Prom.catch((err) => {
+      t.fail('Worker 2.0 run error: ' + JSON.stringify(err));
+    });
+
+    await wf2.signal(unblockSignal);
+
+    await wf1.result();
+    await wf2.result();
+
+    worker1.shutdown();
+    worker2.shutdown();
+    await worker1Prom;
+    await worker2Prom;
+    t.pass();
+  });
+
+  test('Worker versioning client updates', withSkipper, async (t) => {
+    const taskQueue = 'worker-versioning-client-updates-' + uuid4();
+    const conn = t.context.client;
+
+    await conn.taskQueue.updateWorkerBuildIdCompatibility(taskQueue, {
+      operation: 'newIdInNewDefaultSet',
+      buildId: '1.0',
+    });
+    let resp = await conn.taskQueue.getWorkerBuildIdCompatability(taskQueue);
+    assert.equal(resp?.defaultBuildId(), '1.0');
+
+    await conn.taskQueue.updateWorkerBuildIdCompatibility(taskQueue, {
+      operation: 'newCompatibleVersion',
+      buildId: '1.1',
+      existingCompatibleBuildId: '1.0',
+    });
+    resp = await conn.taskQueue.getWorkerBuildIdCompatability(taskQueue);
+    assert.equal(resp?.defaultBuildId(), '1.1');
+
+    // Target nonexistent build ID
+    await t.throwsAsync(
+      conn.taskQueue.updateWorkerBuildIdCompatibility(taskQueue, {
+        operation: 'newCompatibleVersion',
+        buildId: '1.2',
+        existingCompatibleBuildId: 'amnotreal',
+      }),
+      { message: /amnotreal not found/ }
```

Comment:
What's the error class that's thrown here?

Reply:
I think it's just `Error`:

```
Error {
    code: 5,
    details: 'targeted compatible_version amnotreal not found',
    metadata: Metadata {
      internalRepr: Map {
        'content-type' => Array [ … ],
        'grpc-status-details-bin' => Array [ … ],
      },
      options: {},
    },
    message: '5 NOT_FOUND: targeted compatible_version amnotreal not found',
  }
  ```

Reply:
You need to wrap the call in `rethrowGrpcError` like we do in the other clients.

Reply:
K. I added not found but it's not really possible to tie one specific build id to the error in all cases, but that seems fine since the error message itself will have that info and the main thing is matching on type.

Reply:
Add the type to the assertion.